flocker/provision/_install.py

```diff
@@ -34,21 +38,50 @@
 
 ARCHIVE_BUCKET = 'clusterhq-archive'
 
-CLUSTERHQ_REPO = {
-    'fedora-20': "https://s3.amazonaws.com/{archive_bucket}/"
-                 "fedora/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
-                     archive_bucket=ARCHIVE_BUCKET,
-                 ),
-    'centos-7': "https://s3.amazonaws.com/{archive_bucket}/"
-                "centos/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
-                    archive_bucket=ARCHIVE_BUCKET,
-                    ),
-    # FLOC-1828 TODO - use ubuntu rather than ubuntu-testing
-    'ubuntu-14.04': 'https://{archive_bucket}.s3.amazonaws.com/'
-                'ubuntu-testing/14.04/$(ARCH)'.format(
-                    archive_bucket=ARCHIVE_BUCKET
-                    ),
-}
+
+def get_repository_url(distribution, flocker_version):
+    """
+    Return the URL for the repository of a given distribution.
+
+    Note that for ``yum``-using distributions this gives the URL to a package.
+
+    :param bytes distribution: The Linux distribution to get a repository for.
+    :param bytes flocker_version: The version of Flocker to get a repository
+        for.
+
+    :return bytes: The URL pointing to a repository of packages.
+    :raises: ``UnsupportedDistribution`` if the distribution is unsupported.
+    """
+    distro_key_suffix = get_package_key_suffix(flocker_version)
+
+    distribution_to_url = {
+        'fedora-20': "https://{archive_bucket}.s3.amazonaws.com/{key}"
+                     "/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
+                         archive_bucket=ARCHIVE_BUCKET,
+                         key='fedora' + distro_key_suffix,
+                     ),
+        'centos-7': "https://{archive_bucket}.s3.amazonaws.com/"
+                    "{key}/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
+                        archive_bucket=ARCHIVE_BUCKET,
+                        key='centos' + distro_key_suffix,
+                        ),
+        'ubuntu-14.04': 'https://{archive_bucket}.s3.amazonaws.com/'
+                        '{key}/14.04/$(ARCH)'.format(
+                        archive_bucket=ARCHIVE_BUCKET,
+                        key='ubuntu' + distro_key_suffix,
+                        ),
```

Comment:
For apt, it actually makes more sense to provide the entire `deb` line as the parameter, as we might eventually have a deb line in the form `deb https://{archive_bucket}.s3.amazonaws.com/{key} 14.04/${ARCH)`.

However, this could be a follow-up issue (post-1.0).